src/utils/appinsights.ts

```diff
@@ -31,66 +16,13 @@ interface IInsightsRequestData {
  */
 export function initAppInsights(
   instrumentationKey: string,
-  config: Partial<
-    Pick<appInsights.TelemetryClient["config"], "httpAgent" | "httpsAgent">
-  > = {}
+  config: ApplicationInsightsConfig = {}
 ): appInsights.TelemetryClient {
-  appInsights
-    .setup(instrumentationKey)
-    .setAutoDependencyCorrelation(true)
-    .setAutoCollectRequests(true)
-    .setAutoCollectPerformance(true)
-    .setAutoCollectExceptions(true)
-    .setAutoCollectDependencies(true)
-    .setAutoCollectConsole(false)
-    // see https://stackoverflow.com/questions/49438235/application-insights-metric-in-aws-lambda/49441135#49441135
-    .setUseDiskRetryCaching(false)
-    .setSendLiveMetrics(true)
-    .start();
-  appInsights.defaultClient.addTelemetryProcessor(
-    removeQueryParamsPreprocessor
-  );
+  startAppInsights(instrumentationKey, config);
```

Comment:
I've moved this implementation here: https://github.com/pagopa/io-ts-commons/blob/master/src/appinsights.ts#L53
io io-ts-commons since is shared with functions and pagopa-proxy